snap/validate.go

```diff
@@ -664,7 +664,7 @@ func ValidateLayout(layout *Layout, constraints []LayoutConstraint) error {
 		return fmt.Errorf("layout %q uses invalid mount point: must be absolute and clean", layout.Path)
 	}
 
-	for _, path := range []string{"/proc", "/sys", "/dev", "/run", "/boot", "/lost+found", "/media"} {
+	for _, path := range []string{"/proc", "/sys", "/dev", "/run", "/boot", "/lost+found", "/media", "/var/lib/snapd/hostfs"} {
```

Comment:
What about everything else under /var/lib/snapd? Well.. and /var/lib in general?

Reply:
snap-confine only bridges data from `/var/lib/snapd`, `/var/tmp` and `/var/snap` therefore the remaining contents of /var is the read-only core snap. The only exception are all-snap devices where we don't use pivot_root in some cases.

In general though we allow snaps to expose their data into those locations because there's no risk associated with that (they are not getting access to system data and they are not corrupting system data).

Reply:
So to recap: we mount mimics with a tag that shows us this is a mimic, then we only allow tmpfs that has this tag :)

Reply:
We don't disallow /var/lib/snapd but yeah, I think we should. I will add that. As for /var/lib in general, as I said above it is not the real /var/lib, there's no data inside that's useful. In addition the layout only allows you to hide that data and put your stuff there. Is there something you are worried about in particular?

Reply:
Disallowed more things, please comment if there's another thing you'd like to see marked as off-limits.